### ISSUE TITLE
Misinterpreted comment

### ISSUE THREAD

**Issue by brianrthompson** [sgwozdz/jenkinsfile-support#23]
Hi, wonderful plugin if I may add.  Thanks for putting in the time to provide it.

I am reporting a syntax-related error where a command is being mistaken for a comment, which is causing the remainder of the document's text to not be rendered with the nice syntactic colors.

**Use case**
```groovy
def thing = """#!/usr/bin/env bash
do stuff
rm -rf /rpmPackageDir/*.key
do more stuff
```

Everything after the `/*.key` isn't being highlighted, just commented out in gray.

**Comment by sgwozdz** [comment]
Hi @emptyshells,
Thank you for these kind words. I will take a look at this soon.

**Comment by mrwonko** [comment]
I have a similar issue with a `"""` multiline string containing `"unix:///var/run/docker.sock"`—the first `"` is incorrectly interpreted as the string end, according to syntax highlighting, and then the `//` is interpreted as a comment.

It actually looks like you can mix and match any kind of quote. `"foo'` is treated as one string, as is `'bar"` or `"baz'''`.

Might be a separate bug though?

**Comment by fuzzmz** [comment]
@mrwonko running into a similar issue, where having quote interpolation trips up syntax highlight.

For example having something like `script: 'curl -g -s ' + jobURL + ' | python -c "import sys, json; input = json.load(sys.stdin); print input[\'displayName\'][input[\'displayName\'].index(\'v\') + 1:]"'` will make everything after that entry no longer syntax highlighted, as it considers the script string as not being closed.

**Comment by brianrthompson** [comment]
My issue went away when upgrading the plugin to the most recent version, 0.4.3.  I don't think that would resolve the two issues above though.

**Comment by sgwozdz** [comment]
> I have a similar issue with a `"""` multiline string containing `"unix:///var/run/docker.sock"`—the first `"` is incorrectly interpreted as the string end, according to syntax highlighting, and then the `//` is interpreted as a comment.
> 
> It actually looks like you can mix and match any kind of quote. `"foo'` is treated as one string, as is `'bar"` or `"baz'''`.
> 
> Might be a separate bug though?

Looks like separate bug.

**Comment by sgwozdz** [comment]
I've made few improvements in syntax highlighting, do you mind to verify them ?

**Comment by fuzzmz** [comment]
@sgwozdz, looks like your changes fixed the problems for my usecase. Thanks for the quick turnaround!

**Comment by sgwozdz** [comment]
Issue seems to be resolved with version 0.4.4.
